flake.nix

```
@@ -19,173 +20,205 @@
+      installPhase = ''
+        mkdir -p $out
+        find . -type f -maxdepth 1 -name "*.a" -exec install -D {} -t $out/lib \;
+        find . -type f -maxdepth 1 -executable -exec install -D {} -t $out/bin \;
+      '';
+      start-group = optionalString (stdenv.isLinux)
+        ''-Wl,--as-needed -static \
+          -Wl,--start-group'';
+      end-group = optionalString (stdenv.isLinux)
+        ''-static-libgcc -static-libstdc++ \
+          -Wl,--end-group'';
     in
     rec {
-      packages.x86_64-linux.libzip-static = nixpkgs.libzip.overrideAttrs (prev: {
-        cmakeFlags = (prev.cmakeFlags or [ ]) ++ [
-          "-DBUILD_SHARED_LIBS=OFF"
-          "-DBUILD_EXAMPLES=OFF"
-          "-DBUILD_DOC=OFF"
-          "-DBUILD_TOOLS=OFF"
-          "-DBUILD_REGRESS=OFF"
-        ];
-        outputs = [ "out" ];
-      });
-      packages.x86_64-linux.base64 = derivation {
-        name = "updfparser";
-        system = "x86_64-linux";
-        builder = "${nixpkgs.bash}/bin/bash";
-        PATH = "${nixpkgs.coreutils}/bin";
-        args = [
-          "-c"
-          ''
+      packages = {
+        libzip-static = nixpkgs-stat.libzip.overrideAttrs (prev: {
+          cmakeFlags = (prev.cmakeFlags or [ ]) ++ [
+            "-DBUILD_SHARED_LIBS=OFF"
+            "-DBUILD_EXAMPLES=OFF"
+            "-DBUILD_DOC=OFF"
+            "-DBUILD_TOOLS=OFF"
+            "-DBUILD_REGRESS=OFF"
+          ];
+          outputs = [ "out" ];
+        });
+
+        base64 = mkDerivation {
+          name = "base64";
+          src = "${base64-src}";
+
+          phases = [ "unpackPhase" "installPhase" ];
+
+          installPhase = ''
             mkdir -p $out/include/base64
-            cp ${base64-src}/Base64.h $out/include/base64/Base64.h
-          ''
-        ];
-      };
-      packages.x86_64-linux.updfparser = derivation {
-        name = "updfparser";
-        system = "x86_64-linux";
-        builder = "${nixpkgs.bash}/bin/bash";
-        PATH = "${nixpkgs.coreutils}/bin";
-        args = [
-          "-c"
-          ''
-            ${cxx} \
-              -c ${updfparser-src}/src/*.cpp \
-              -I ${updfparser-src}/include \
+            cp Base64.h $out/include/base64/Base64.h
+          '';
+        };
+
+        updfparser = mkDerivation {
+          inherit installPhase;
+          name = "updfparser";
+          src = "${updfparser-src}";
+
+          buildPhase = ''
+            ${cc} \
+              -c src/*.cpp \
+              -I include \
               ${obj-flags}
-            mkdir -p $out/lib
-            ${ar} crs $out/lib/libupdfparser.a *.o
-          ''
-        ];
-      };
-      packages.x86_64-linux.gourou = derivation {
-        name = "gourou";
-        system = "x86_64-linux";
-        builder = "${nixpkgs.bash}/bin/bash";
-        PATH = "${nixpkgs.coreutils}/bin";
-        args = [
-          "-c"
-          ''
-            ${cxx} \
+            ${ar} crs lib$name.a *.o
```

For each of these, make the set recursive (i.e. `rec {`) and change it like this
```suggestion
            ${ar} crs lib${name}.a *.o
```
because nix is more intuitive than bash